sklearn/linear_model/_sag_fast.pyx.tp

```diff
@@ -365,52 +347,63 @@ def sag{{name_suffix}}(SequentialDataset{{name_suffix}} dataset,
                          % loss_function)
 
     if prox:
-        cumulative_sums_prox_array = np.empty(n_samples,
-                                              dtype={{np_type}}, order="c")
-        cumulative_sums_prox = <{{c_type}}*> cumulative_sums_prox_array.data
+        cumulative_sums_prox = np.empty(n_samples, dtype={{np_type}}, order="c")
+        cumulative_sums_prox_ptr = &cumulative_sums_prox[0]
     else:
-        cumulative_sums_prox = NULL
+        cumulative_sums_prox = None
+        cumulative_sums_prox_ptr = NULL
 
     with nogil:
         start_time = time(NULL)
         for n_iter in range(max_iter):
             for sample_itr in range(n_samples):
                 # extract a random sample
-                sample_ind = dataset.random(&x_data_ptr, &x_ind_ptr, &xnnz,
-                                              &y, &sample_weight)
+                sample_ind = dataset.random(&x_data_ptr, &x_ind_ptr, &xnnz, &y, &sample_weight)
 
                 # cached index for gradient_memory
                 s_idx = sample_ind * n_classes
 
                 # update the number of samples seen and the seen array
-                if seen[sample_ind] == 0:
+                if seen_init[sample_ind] == 0:
                     num_seen += 1
-                    seen[sample_ind] = 1
+                    seen_init[sample_ind] = 1
 
                 # make the weight updates
                 if sample_itr > 0:
-                   status = lagged_update{{name_suffix}}(weights, wscale, xnnz,
-                                                  n_samples, n_classes,
-                                                  sample_itr,
-                                                  cumulative_sums,
-                                                  cumulative_sums_prox,
-                                                  feature_hist,
-                                                  prox,
-                                                  sum_gradient,
-                                                  x_ind_ptr,
-                                                  False,
-                                                  n_iter)
+                   status = lagged_update{{name_suffix}}(
+                       weights,
+                       wscale,
+                       xnnz,
+                       n_samples,
+                       n_classes,
+                       sample_itr,
+                       &cumulative_sums[0],
+                       cumulative_sums_prox_ptr,
+                       &feature_hist[0],
+                       prox,
+                       sum_gradient,
+                       x_ind_ptr,
+                       False,
+                       n_iter
+                   )
```

Comment:
Is it possible to use key-word arguments for calls that are being modified in this PR, like this one?